ethers-solc/src/artifacts/ast/lowfidelity.rs

```diff
@@ -24,6 +24,14 @@ pub struct Ast {
     pub other: BTreeMap<String, serde_json::Value>,
 }
 
+impl Ast {
+    // TODO: find a better name
+    pub fn to_typed(self) -> SourceUnit {
+        let data = serde_json::to_string(&self).unwrap();
+        serde_json::from_str(&data).unwrap()
+    }
```

Comment:
@mattsse shouldn't we be converting manually from the internal types, instead of ser/de'ing? or am i missing something?

Reply:
That would probably imply to write (or reuse, maybe in coverage) a node walker to convert (de-reserialize) every node so not sure if the performance is too bad doing that ?